File: _append_dubbel_data.py
#!/usr/bin/env python3
"""
Append Belgian Dubbel boost recipes to ML dataset
"""
import json
import sys

def main():
    # Load existing dataset
    print("Loading existing ML dataset...")
    with open('_ml_dataset.json', 'r', encoding='utf-8') as f:
        dataset = json.load(f)

    # Load new Dubbel recipes
    print("Loading Dubbel boost recipes...")
    with open('_dubbel_boost_recipes.json', 'r', encoding='utf-8') as f:
        new_recipes = json.load(f)

    # Get current recipe count
    original_count = len(dataset.get('data', []))
    print(f"Original dataset: {original_count} recipes")

    # Find 'data' key in dataset structure
    if 'data' in dataset:
        data_key = 'data'
    else:
        # Check if recipes are stored directly or under different key
        for key in dataset:
            if isinstance(dataset[key], list) and len(dataset[key]) > 0:
                if isinstance(dataset[key][0], dict) and 'label_slug' in dataset[key][0]:
                    data_key = key
                    break
        else:
            print("ERROR: Could not find recipes data in dataset")
            sys.exit(1)

    # Append new recipes
    dataset[data_key].extend(new_recipes)

    # Update metadata
    new_count = len(dataset[data_key])
    print(f"After boost: {new_count} recipes (+{new_count - original_count})")

    if '_meta' in dataset:
        dataset['_meta']['count'] = new_count
        dataset['_meta']['version'] = 'dubbel_boost_v1'
        dataset['_meta']['source'] = dataset['_meta'].get('source', '') + ' + dubbel_boost_15'

    # Write back
    print("Writing updated dataset...")
    with open('_ml_dataset.json', 'w', encoding='utf-8') as f:
        json.dump(dataset, f, indent=2)

    print("✅ Dubbel data boost complete!")
    print(f"Belgian Dubbel count should be: {11 + 15} = 26 recipes")

if __name__ == '__main__':
    main()

File: _append_dubbel_data.js
#!/usr/bin/env node
/**
 * Append Belgian Dubbel boost recipes to ML dataset
 */
const fs = require('fs');

function main() {
    try {
        // Load existing dataset
        console.log("Loading existing ML dataset...");
        const dataset = JSON.parse(fs.readFileSync('_ml_dataset.json', 'utf8'));

        // Load new Dubbel recipes
        console.log("Loading Dubbel boost recipes...");
        const newRecipes = JSON.parse(fs.readFileSync('_dubbel_boost_recipes.json', 'utf8'));

        // Find data array in dataset
        let dataKey = null;
        if ('data' in dataset) {
            dataKey = 'data';
        } else {
            // Search for array of recipes
            for (const key in dataset) {
                if (Array.isArray(dataset[key]) && dataset[key].length > 0) {
                    if (typeof dataset[key][0] === 'object' && 'label_slug' in dataset[key][0]) {
                        dataKey = key;
                        break;
                    }
                }
            }
        }

        if (!dataKey) {
            console.error("ERROR: Could not find recipes data in dataset");
            process.exit(1);
        }

        const originalCount = dataset[dataKey].length;
        console.log(`Original dataset: ${originalCount} recipes`);

        // Append new recipes
        dataset[dataKey] = dataset[dataKey].concat(newRecipes);

        const newCount = dataset[dataKey].length;
        console.log(`After boost: ${newCount} recipes (+${newCount - originalCount})`);

        // Update metadata
        if (dataset._meta) {
            dataset._meta.count = newCount;
            dataset._meta.version = 'dubbel_boost_v1';
            dataset._meta.source = (dataset._meta.source || '') + ' + dubbel_boost_15';
        }

        // Write back
        console.log("Writing updated dataset...");
        fs.writeFileSync('_ml_dataset.json', JSON.stringify(dataset, null, 2));

        console.log("✅ Dubbel data boost complete!");
        console.log(`Belgian Dubbel count should be: 11 + 15 = 26 recipes`);

    } catch (error) {
        console.error("ERROR:", error.message);
        process.exit(1);
    }
}

main();